From 7681e57a60e5d54b2e319db5b82428a0fe048255 Mon Sep 17 00:00:00 2001
From: benaryorg <binary@benary.org>
Date: Sat, 17 May 2025 01:26:12 +0000
Subject: [PATCH] analyze: redirect runtime directory to tmp

By redirecting to runtime directory to a temporary directory the tool is able to run in a build process/sandbox which may not have access to */run*.

Signed-off-by: benaryorg <binary@benary.org>
---
 src/analyze/analyze-verify-util.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/analyze/analyze-verify-util.c b/src/analyze/analyze-verify-util.c
index 4e034ae7ad..1500df5fbe 100644
--- a/src/analyze/analyze-verify-util.c
+++ b/src/analyze/analyze-verify-util.c
@@ -12,6 +12,7 @@
 #include "manager.h"
 #include "pager.h"
 #include "path-util.h"
+#include "sd-path.h"
 #include "string-table.h"
 #include "strv.h"
 #include "unit-name.h"
@@ -306,6 +307,10 @@ int verify_units(
         if (r < 0)
                 return log_error_errno(r, "Failed to initialize manager: %m");
 
+        r = sd_path_lookup(SD_PATH_TEMPORARY, NULL, &m->prefix[EXEC_DIRECTORY_RUNTIME]);
+        if (r < 0)
+                return log_error_errno(r, "Failed to use temporary directory as runtime directory: %m");
+
         log_debug("Starting manager...");
 
         r = manager_startup(m, /* serialization= */ NULL, /* fds= */ NULL, root);
-- 
2.47.2

